Bermuda/Bermuda/Bat.cpp

```diff
@@ -50,6 +51,19 @@ MovableEntity(id, spawnPoint->getX(), spawnPoint->getY())
 
 void Bat::update(double dt)
 {
+	double diffX = PlayState::Instance()->getPlayer()->getCenterX() - this->getCenterX();
+	double diffY = PlayState::Instance()->getPlayer()->getCenterY() - this->getCenterY();
+	double distanceFromPlayer = sqrt((diffX * diffX) + (diffY * diffY));
+
+	if (this->m_pStateMachine->getCurrentState() == WanderAround::Instance() && distanceFromPlayer <= 150)
+	{
+		this->m_pStateMachine->changeState(FleeingState::Instance());
+	}
+	else if (this->m_pStateMachine->getCurrentState() == FleeingState::Instance() && distanceFromPlayer >= 300)
+	{
+		this->m_pStateMachine->changeState(WanderAround::Instance());
+	}
+
```

Comment:
Probeer het wisselen van states ín de state zelf te zetten (dus bv, de eerste "if" kan dan in WanderAround instance komen, door de distanceFromPlayer uit Entity op te halen (in dit geval dus de Bat). Zo ja, switch dan naar de FleeState, vanuit de WanderAround state. Zelfde geldt dan voor de fleeingstate, maar dan bij distance > 300)